### PR TITLE
Improve attachments streaming with a new AttachmentsStreamingPool class

### PR DESCRIPTION
**Pull request by patricijabrecko**
## Summary

This pull request introduces a new `AttachmentsStreamingPool` class to handle streaming attachments with improved concurrency and error handling. It replaces the previous reducer and iterator-based approach in the `WorkerAdapter` class. 

## Connected Issues

https://app.devrev.ai/devrev/works/ISS-174249

## Craftsmanship, Integrity, and Devil’s Advocacy

- [ ] Testing: Negative test cases: null or default values, crash and fault injection tests
- [ ] Testing: Boundary conditions: rolling upgrades, denial-of-service, etc.
- [ ] Testing: Fixing a few existing — flaky or permanently-broken — test cases
- [ ] Observing: Detailed error codes so machines can understand
- [ ] Observing: Adding superior metrics for future debugging
- [ ] Observing: Tracing the hairiest pathways for field serviceability
- [ ] Training: KnowledgeOps update? So AI always works.

### CODE REVIEW

**Review by radovanjorgic — CHANGES_REQUESTED** [review]

**Review by radovanjorgic — CHANGES_REQUESTED** [review]
Also, please use more descriptive and specific PR title since release notes are generated from this.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@devrev/ts-adaas",
-  "version": "1.7.2-beta.0",
+  "version": "1.8.0-beta.0",
```
[the remark this answers is not in the document]
Comment (patricijabrecko) [reply package.json:3]: It wasn't changed manually, when I released the beta version directly from this branch, it updated it here :) But as we release from main, it will also be changed there so I think it shouldn't be a problem?

**Review by gasperzgonec — COMMENTED** [review]
To improve code readability, I'd suggest we move `this.state.toDevRev.attachmentsMetadata` to a shorter variable.
As it's an object, the assignment is a reference, and the changes are propagated to the `this.state`.

We already checked that the value is not null, so we can assign it here without issues.
---
The `here` refers to the line 997 (right before ` // Loop through the attachments metadata artifact IDs`) of code located [here](https://github.com/devrev/adaas-sdk/pull/56/files/662575b3ae0b14dc1d65f08c0e9ba4156450d8c4#diff-44e201ba7ec3aebbd55ee2025b44e52a45ac68c95b1af55e81c38cfea0aaa655R847) (can't comment on the line, as it's not a change)

File: src/attachments-streaming/attachments-streaming-pool.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    await Promise.all(initialPromises);
+
+    if (this.delay) {
+      return { delay: this.delay };
+    }
+
+    return {};
+  }
+
+  async startPoolWorker() {
```
[the remark this answers is not in the document]
Comment (patricijabrecko) [reply src/attachments-streaming/attachments-streaming-pool.ts]: True, I renamed it and removed all usage of "worker" in logs and comments

**Comment by gasperzgonec** [comment]
LGTM! No more comments 👍

**Review by radovanjorgic — APPROVED** [review]
